Backends/PDF/sheet.lisp

```diff
@@ -0,0 +1,182 @@
+;;; License as published by the Free Software Foundation; either
+;;; version 2 of the License, or (at your option) any later version.
+;;;
+;;; This library is distributed in the hope that it will be useful,
+;;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;;; Library General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU Library General Public
+;;; License along with this library; if not, write to the
+;;; Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+;;; Boston, MA  02111-1307  USA.
+
+
+(in-package :clim-pdf)
+
+(defmacro with-output-to-pdf-stream ((stream-var file-stream
+                                             &rest options)
+                                            &body body)
+  (let ((cont (gensym)))
+    `(flet ((,cont (,stream-var)
+              ,@body))
+       (declare (dynamic-extent #',cont))
+       (invoke-with-output-to-pdf-stream #',cont
+                                                ,file-stream ,@options))))
+
+(defun invoke-with-output-to-pdf-stream (continuation
+                                         file-stream
+                                         &key device-type
+                                              multi-page scale-to-fit
+                                              (orientation :portrait)
+                                              header-comments)
+  (let* ((port (find-port :server-path `(:pdf :stream ,file-stream)))
+         (stream (make-clim-pdf-stream file-stream port device-type
+                                  multi-page scale-to-fit
+                                  orientation header-comments))
+         translate-x translate-y)
+    (declare (ignore translate-x translate-y))
+    (unwind-protect
+         (progn
+           (with-output-recording-options (stream :record t :draw nil)
+             (funcall continuation stream)
+             (new-page stream))
+           (with-slots (file-stream title for orientation paper) stream
+             (let ((flexi-stream
+                    (flexi-streams:make-flexi-stream
+                     file-stream
+                     :external-format :latin-1)))
+               (with-output-recording-options (stream :draw t :record nil)
+                 (let ((pdf:*compress-streams* nil))
+                   (pdf:with-document ()
+                     (pdf:with-page ()
+                       (let ((last-page (first (pdf-pages stream))))
+                         (dolist (page (reverse (pdf-pages stream)))
+                           (replay page stream)
+                           (unless (eql page last-page)
+                             (emit-new-page stream)))))
+                     (pdf:write-document flexi-stream)))))))
+      (destroy-port port))))
+
+
+(defun start-page (stream)
+  (with-slots (file-stream current-page transformation) stream
+    ))
+
```

Comment:
maybe put nil here to indicate that nothing is done? (not sure about that style "fix", just suggestion)